op-node/rollup/derive/engine_update.go

```diff
@@ -157,6 +178,10 @@ func confirmPayload(ctx context.Context, log log.Logger, eng ExecEngine, fc eth.
 			return nil, BlockInsertTemporaryErr, fmt.Errorf("failed to make the new L2 block canonical via forkchoice: %w", err)
 		}
 	}
+	// at this point the block is either inserted, or not valid, so clear it from the async gossiper
+	if agossip != nil {
```

Comment:
Probably should have asked this on the first pass sorry, but would we be better off using an interface for the `agossip` type here and setting it to a no-op implementation if its not in use rather than needing to do `!= nil` checks all the time?

Reply:
Yeah, we can do that! Only one path that feeds in a `nil` so it should be easy to handle.